packages/meditrak-app/app/assessment/QrCodeScreen.js

```diff
@@ -0,0 +1,162 @@
+                qrCodeImgRefs.current[index] = ref;
+              }}
+              width={300}
+              qrCodeContents={data}
+              humanReadableId={name}
+            />
+            <View style={localStyles.buttonsContainer}>
+              <Button
+                style={[
+                  localStyles.smallButton,
+                  localStyles.transparentButton,
+                  localStyles.downloadButton,
+                ]}
+                onPress={() => {
+                  downloadQrCode(
+                    qrCodeImgRefs.current[index],
+                    name,
+                    () => displayDownloadMessage('Download successful'),
+                    errorMessage => displayDownloadMessage(`Download failed: ${errorMessage}`),
+                  );
+                }}
+                title="Download"
+                textStyle={[localStyles.buttonLabel, localStyles.transparentButtonLabel]}
+              />
+              <Button
+                style={localStyles.smallButton}
+                onPress={() => {
+                  shareQrCode(qrCodeImgRefs.current[index], name);
+                }}
+                title="Share"
+                textStyle={[localStyles.buttonLabel, localStyles.shareButtonLabel]}
+              />
+            </View>
+            <Divider style={localStyles.divider} />
+          </View>
+        ))}
+
+        <View style={localStyles.buttonsContainer}>
+          <Button
+            style={[localStyles.transparentButton, localStyles.closeButton]}
+            title="Close"
+            onPress={onClose}
+            textStyle={[localStyles.buttonLabel, localStyles.transparentButtonLabel]}
+          />
+        </View>
+      </ScrollView>
+    </TupaiaBackground>
+  );
+};
+
+QrCodeScreenComponent.navigationOptions = ({ navigation }) => ({
+  headerTitle: navigation.state.params.title,
+});
+
+const localStyles = StyleSheet.create({
+  heading: {
+    paddingLeft: 20,
+    marginVertical: 30,
+    fontSize: THEME_FONT_SIZE_ONE,
+    flex: 0,
+    textAlign: 'left',
+  },
+  qrCodeContainer: {
+    display: 'flex',
+    alignItems: 'center',
+    marginBottom: 10,
+  },
+  buttonsContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  divider: {
+    width: 300,
+    marginTop: 20,
+    marginBottom: 20,
+  },
+  closeButton: {
+    marginBottom: 20,
+    width: 300,
+  },
+  smallButton: { marginTop: 20, marginBottom: 5, width: 142 },
+  downloadButton: { marginRight: 16 },
+  buttonLabel: {
+    fontWeight: 'bold',
+  },
+  transparentButton: {
+    borderWidth: 1,
+    borderColor: THEME_COLOR_ONE,
+    backgroundColor: 'transparent',
+  },
+  transparentButtonLabel: {
+    color: THEME_COLOR_ONE,
+  },
+});
+
+const mapStateToProps = () => ({});
+
+const mapDispatchToProps = dispatch => ({
+  displayDownloadMessage: message => dispatch(addMessage('qr_code_download', message)),
+});
```

Comment:
Had to use redux connect here to be able to display the download success message.